### PR TITLE
Add Github Action Step To Create a change-set When The Helm Chart Is Pushed

### PR DESCRIPTION
**Pull request by maximillianbrain1**
Towards ENG-1227

Adds a step to create an AWS Marketplace change-set to update our product offering each time the Helm Chart is being pushed to the AWS Marketplace Registry. 

Example change-set output:

```
aws marketplace-catalog start-change-set --change-set-name Publish version v1.1.4 --catalog AWSMarketplace --region us-east-1 --change-set [
  {
    "ChangeType": "AddDeliveryOptions",
    "Entity": {
      "Identifier": "prod-m7tlrvfq6yjzu",
      "Type": "ContainerProduct@1.0"
    },
    "Details": "{\"Version\":{\"ReleaseNotes\":\"For detailed release notes, view our release notes for this version at: https://github.com/ksoclabs/ksoc-plugins-helm-chart\",\"VersionTitle\":\"Chart Release v1.1.4\"},\"DeliveryOptions\":[{\"Details\":{\"HelmDeliveryOptionDetails\":{\"CompatibleServices\":[\"EKS\"],\"ContainerImages\":[\"709825985650.dkr.ecr.us-east-1.amazonaws.com/ksoc-labs/falco-no-driver:0.36.2\",\"709825985650.dkr.ecr.us-east-1.amazonaws.com/ksoc-labs/falcoctl:0.6.2\\\\nus.gcr.io/ksoc-public/ksoc-bootstrapper:v1.1.2\\\\nus.gcr.io/ksoc-public/ksoc-guard:v1.1.2\\\\nus.gcr.io/ksoc-public/ksoc-sbom:v1.1.4\\\\nus.gcr.io/ksoc-public/ksoc-sync:v1.1.2\\\\nus.gcr.io/ksoc-public/ksoc-watch:v1.1.2\\\\n\"],\"Description\":\"The KSOC plugins are deployed in your EKS cluster with our Helm chart. The chart is open source and can be viewed here\",\"HelmChartUri\":\"709825985650.dkr.ecr.us-east-1.amazonaws.com/ksoc-labs/ksoc-plugins:1.1.4\",\"QuickLaunchEnabled\":false,\"UsageInstructions\":\"Install with Helm or EKS Addon\",\"ReleaseName\":\"ksoc-plugins\",\"Namespace\":\"ksoc\",\"OverrideParameters\":[{\"Key\":\"ksoc.clusterName\",\"Metadata\":{\"Obfuscate\":false,\"Label\":\"KSOC Cluster Name\",\"Description\":\"The name of the cluster you wish to give to within the KSOC Platform\"}},{\"Key\":\"ksoc.base64AccessKeyId\",\"Metadata\":{\"Obfuscate\":true,\"Label\":\"Base64 Access Key\",\"Description\":\"The base64 encoded acess key given through the UI or the API of KSOC\"}},{\"Key\":\"ksoc.base64SecretKey\",\"Metadata\":{\"Obfuscate\":true,\"Description\":\"The base64 encoded secrt key given through the UI or the API of KSOC\",\"Label\":\"Base64 Secret Key\"}}]}},\"DeliveryOptionTitle\":\"KSOC\"}]}"
  }
]
```

### CODE REVIEW

File: scripts/create_changeset.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
```
Comment (pancernik) [inline scripts/create_changeset.sh:2]: Can we do `set -euo pipefail`?

File: scripts/create_changeset.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    repo=$(yq e ".${image}.image.repository" "$VALUES_FILE_PATH")
+    tag=$(yq e ".${image}.image.tag" "$VALUES_FILE_PATH")
+
+    # Check if both repository and tag are not null or empty
+    if [ -n "$repo" ] && [ "$repo" != "null" ] && [ -n "$tag" ] && [ "$tag" != "null" ]; then
+        images+="${repo}:${tag}\n"
+    fi
+done
+
+# Appendthe images found in values.yaml with our Falco images
```
Comment (pancernik) [inline scripts/create_changeset.sh:32]: Nit: `Append the`

Comment (pancernik) [reply scripts/create_changeset.sh:32]: Actually, do we have to mention Falco here?

Comment (maximillianbrain1) [reply scripts/create_changeset.sh:32]: We do. 😞 All images have to be hosted within AWS, so I manually copied them over for now. When Falco gets replaced, we can remove it.

File: scripts/create_changeset.sh (hunk 1 above)
Comment (maximillianbrain1) [reply scripts/create_changeset.sh:3]: We can. I added it.

**Review by mateuszdyminski — APPROVED** [review]

**Pull request merged**